evil-states.el

```diff
@@ -99,6 +99,8 @@ commands opening a new line."
                  evil-open-below
                  evil-append
                  evil-append-line
+                 evil-change
```

Comment:
I understand the `evil-change-line` addition, but not `evil-change` if I do `c$<escape>` when at the start of a line, this is an `evil-change` but vim doesn't remove the indentation spaces..?

Reply:
- `evil-change` is needed for  `cc <escape>` to remove indentation spaces
  - As a side effect, this also makes  `c $ <escape>` remove indentation spaces
- `evil-change-line` is needed for `C <escape>` to remove  indentation spaces
  - But on my vim `C <escape>` does not remove the indentation spaces

I don't know how to make both `cc <escape>` and `c $ <escape>` behave the same as vim. But the minor difference is acceptable for me. 

My actual need is to automatically remove trailing spaces at the cursor line when I leave insert state.

Reply:
Ah yes, I understand. I've noticed that vi, vim and neovim all behave differently. So evil doesn't really have a solid example to follow. I agree that `c$<escape>` is probably acceptable. I won't merge this without tests, so either you'll have to add them, or you'll have to wait for me to add them (sometime next week I hope).